processing/src/main/java/io/druid/query/BySegmentQueryRunner.java

```diff
@@ -61,7 +62,11 @@ public BySegmentQueryRunner(
                   new BySegmentResultValueClass<T>(
                       results,
                       segmentIdentifier,
-                      query.getIntervals().get(0)
+                      Iterables.getOnlyElement(
+                          Iterables.getOnlyElement(query.getDataSources())
```

Comment:
I think ```getDataSources()``` is mostly used internally like ```ServerManager``` or ```QueryManager```, and developers should keep in mind that a query can have multiple data sources when they modify the codes where do something with data sources. 

There are some exceptions like ```BySegmentQueryRunner```, ```SinkQuerySegmentWalker```, and ```SpecificSegmentQueryRunner``` which expect a query must have a single data source. I think this will be rare, and would like to keep the current implementation.